### PR TITLE
add dry run and output

### PR DESCRIPTION
**Pull request by mmulholla**
https://issues.redhat.com/browse/APPSVC-1359

Added dry-run flag to create and join commands. For example output:
```
[INFO] (dry run) SUCCESS: create of Namespace primaza-alice was successful
[INFO] (dry run) SUCCESS: create of CustomResourceDefinition clusterenvironments.primaza.io was successful
[INFO] (dry run) SUCCESS: create of CustomResourceDefinition registeredservices.primaza.io was successful
[INFO] (dry run) SUCCESS: create of CustomResourceDefinition servicebindings.primaza.io was successful
[INFO] (dry run) SUCCESS: create of CustomResourceDefinition servicecatalogs.primaza.io was successful
[INFO] (dry run) SUCCESS: create of CustomResourceDefinition serviceclaims.primaza.io was successful
[INFO] (dry run) SUCCESS: create of CustomResourceDefinition serviceclasses.primaza.io was successful
[ERROR] (dry run) FAILED: create of ServiceAccount primaza-controller-manager: Exception: namespaces "primaza-alice" not found
[ERROR] (dry run) FAILED: create of Role primaza-claimer: Exception: namespaces "primaza-alice" not found
[ERROR] (dry run) FAILED: create of Role primaza-leader-election-role: Exception: namespaces "primaza-alice" not found
[ERROR] (dry run) FAILED: create of Role primaza-manager-role: Exception: namespaces "primaza-alice" not found
[ERROR] (dry run) FAILED: create of Role primaza-reporter: Exception: namespaces "primaza-alice" not found
[ERROR] (dry run) FAILED: create of RoleBinding primaza-leader-election-rolebinding: Exception: namespaces "primaza-alice" not found
[ERROR] (dry run) FAILED: create of RoleBinding primaza-manager-rolebinding: Exception: namespaces "primaza-alice" not found
[ERROR] (dry run) FAILED: create of ConfigMap primaza-manager-config: Exception: namespaces "primaza-alice" not found
[ERROR] (dry run) FAILED: create of Service primaza-webhook-service: Exception: namespaces "primaza-alice" not found
[ERROR] (dry run) FAILED: create of Deployment primaza-controller-manager: Exception: namespaces "primaza-alice" not found
[ERROR] (dry run) FAILED: create of Certificate primaza-serving-cert: Exception: namespaces "primaza-alice" not found
[ERROR] (dry run) FAILED: create of Issuer primaza-selfsigned-issuer: Exception: namespaces "primaza-alice" not found
[INFO] (dry run) SUCCESS: create of ValidatingWebhookConfiguration primaza-validating-webhook-configuration was successful
Dry run Primaza tenant install complete
```

Had to skip some checks in the code which rely in the resources being created.

Also added output as yaml option, if specified the only output to stdout is a yaml array of resources created which is similar to kubectl output. If a resource is modified to obscure secret data a warning is output to stderr, for example:
`WARNING:Secret primaza-tkn-primaza-alice-cluster1dev: "kubeconfig" attribute modified to hide secrets` 

Also updated readme and added new tests. The new test also added to PR checks.

### CODE REVIEW

File: scripts/src/primazactl/utils/settings.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import yaml
+import sys
+from primazactl.utils import logger
+
+dry_run = False
```
Comment (filariow) [inline scripts/src/primazactl/utils/settings.py]: I'd propose to allow `dry_run` to have the following values:
* `None`: no dry-run, execute the command
* `server`: behave as implemented in this PR, check with the target cluster if the command can be performed
* `client`: won't execute anything, just return. This won't require a running target cluster and can be used for printing the yamls and for validation

File: scripts/src/primazactl/utils/settings.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+output_yaml = False
```
Comment (filariow) [inline scripts/src/primazactl/utils/settings.py]: I'd propose to have an `output` variable that can be `None` or `yaml` as of now. In future we may want to add `json` or other formatters.

**Review by filariow — CHANGES_REQUESTED** [review]
some nits

File: scripts/src/primazactl/utils/settings.py (hunk 2 above)
Comment (mmulholla) [reply scripts/src/primazactl/utils/settings.py]: I changed to be more to your liking. would be a minor update for a adding a new type either way.  Adding json would be about a 5 line change to the code, but a bit more for adding a test.

**Review by filariow — APPROVED** [review]
lgtm

**Pull request merged**